cli/flox-rust-sdk/src/providers/nix.rs

```diff
@@ -0,0 +1,31 @@
+use std::path::PathBuf;
+use std::process::Command;
+use std::sync::LazyLock;
+
+static NIX_BIN: LazyLock<PathBuf> = LazyLock::new(|| {
+    std::env::var("NIX_BIN")
+        .unwrap_or_else(|_| env!("NIX_BIN").to_string())
+        .into()
+});
+
+/// Returns a `Command` for `nix` with a default set of features enabled.
+pub fn nix_base_command() -> Command {
+    let mut command = Command::new(&*NIX_BIN);
+    command.args([
+        "--option",
+        "extra-experimental-features",
+        "nix-command flakes",
+    ]);
+    command
+}
```

Comment:
Great, happy to see this factored out a bit.

My plan was to have this look a bit more like the git provider, i.e have  a bit more config options (pure eval, store?, flake arguemtns reg. locking...) and common output handling (e.g. automatically printing the executed command and `tap`ing the outputs since that's what we do in all the commands anyway)

our nix uses are arguably diverse enough that that might not help a lot right now.

Reply:
Yeah, there's not a lot of other commonality at the moment, so I think we should defer any more abstractions over a plain `Command` until we need better output tapping in those places.